### DISCUSSION TITLE
Game crashing at startup when database cannot be opened

### DISCUSSION THREAD

**Discussion by adryzz** [ppy/osu#16222] [Q&A] [answered]
Yesterday i imported my data from stable, with a couple of beatmaps/skins not importing properly (as expected). Some aspire maps didn't import and i expected that from the osu beatmap file format change.

![image](https://user-images.githubusercontent.com/46694241/147204418-b264152e-f2f4-4c43-b7dc-36cdb390812c.png)

I left for a bit, and then i came back with a notification saying something along the lines of "524 beatmaps imported successfully", and i thought all the other beatmaps were invalid. I clicked on the notification and the game crashed.
The game wouldn't start again, and i thought it was something similiar to #6568 but by looking at the logs, it crashes with a stack trace saying it can't open the database file.

[performance.log](https://github.com/ppy/osu/files/7767505/performance.log)
[performance-audio.log](https://github.com/ppy/osu/files/7767506/performance-audio.log)
[performance-draw.log](https://github.com/ppy/osu/files/7767507/performance-draw.log)
[performance-input.log](https://github.com/ppy/osu/files/7767508/performance-input.log)
[performance-update.log](https://github.com/ppy/osu/files/7767509/performance-update.log)
[runtime.log](https://github.com/ppy/osu/files/7767510/runtime.log)
[database.log](https://github.com/ppy/osu/files/7767503/database.log)
[network.log](https://github.com/ppy/osu/files/7767504/network.log)

Stack trace wasn't in the logs, so i copied it from the terminal window
[stacktrace.txt](https://github.com/ppy/osu/files/7767563/stacktrace.txt)

(I assume i'll have to send the database as well, but client.db is 0 bytes for some reason)

()
**Expected behavior:**
Game importing maps and starting up properly or game starting up with an error and not opening the database file.

**What actually happened:**
Startup crash

**Reproducibility: Very Reproducible if using my database file**

**osu!lazer version & specifics**
osu 2021.1218.0.0

Arch Linux 5.15.10-zen1-1-zen

**Comment by peppy** [comment, marked as answer]
Please delete the database file and try again.

**Reply by adryzz** [reply]
That makes the game start, i guess. I mean... no more beatmaps tho (not that after the import fail there were any left)

**Reply by adryzz** [reply]
I'll start the import again and grab the logs from the import instead to see what's happening there

**Reply by smoogipoo** [reply]
Hopefully things will be more robust after we completely switch to the new database backend.

**Reply by adryzz** [reply]
Usually exceptions are caught at runtime with a notification, so i was kinda of shocked to see a crash

**Reply by bdach** [reply]
Well not if a database file just gets truncated to 0 bytes for some reason...
